specifyweb/frontend/js_src/lib/components/QueryBuilder/Formatter.tsx

```diff
@@ -140,9 +140,24 @@ function FormatSelect({
   readonly onChange: ((formatter: string | undefined) => void) | undefined;
 }): JSX.Element | null {
   const [formatterSelectIsOpen, setFormatterSelect] = React.useState(false);
-
+  const [selectedIndex, setSelectedIndex] = React.useState<number | null>(null);
   const id = useId('formatters-selection');
 
+  const compositeValue = React.useMemo(() => {
+    if (availableFormatters && availableFormatters.length > 0) {
+      if (selectedIndex !== null && availableFormatters[selectedIndex]) {
+        return `${availableFormatters[selectedIndex].name}|||${selectedIndex}`;
```

Comment:
Since this string format is used in multiple places, can you make a function that accepts two params `name` and `index` and returns this generated string? It would help not only with reusability but also to document what the string represents